src/content_mapblock.cpp

```diff
@@ -525,1633 +598,1395 @@ static TileSpec getSpecialTile(const ContentFeatures &f,
-					If neighbor is not liquid, lower border of face is
-					lower border of node
-				*/
-				else
-				{
-					vertices[0].Pos.Y = -0.5*BS;
-					vertices[1].Pos.Y = -0.5*BS;
-				}
+void MapblockMeshGenerator::drawGlasslikeNode()
+{
+	TileSpec tile = getNodeTile(n, p, v3s16(0,0,0), data);
 
-				for(s32 j=0; j<4; j++)
-				{
-					if(dir == v3s16(0,0,1))
-						vertices[j].Pos.rotateXZBy(0);
-					if(dir == v3s16(0,0,-1))
-						vertices[j].Pos.rotateXZBy(180);
-					if(dir == v3s16(-1,0,0))
-						vertices[j].Pos.rotateXZBy(90);
-					if(dir == v3s16(1,0,-0))
-						vertices[j].Pos.rotateXZBy(-90);
-
-					// Do this to not cause glitches when two liquids are
-					// side-by-side
-					/*if(neighbor_is_same_liquid == false){
-						vertices[j].Pos.X *= 0.98;
-						vertices[j].Pos.Z *= 0.98;
-					}*/
-
-					if (data->m_smooth_lighting)
-						vertices[j].Color = blendLight(frame, vertices[j].Pos, current_tile->color);
-					vertices[j].Pos += intToFloat(p, BS);
-				}
+	u16 l = getInteriorLight(n, 1, nodedef);
+	video::SColor c = encode_light_and_color(l, tile.color,
+		f->light_source);
+	for(u32 j=0; j<6; j++)
+	{
+		// Check this neighbor
+		v3s16 dir = g_6dirs[j];
+		v3s16 n2p = blockpos_nodes + p + dir;
+		MapNode n2 = data->m_vmanip.getNodeNoEx(n2p);
+		// Don't make face if neighbor is of same type
+		if(n2.getContent() == n.getContent())
+			continue;
+		video::SColor c2=c;
+		if(!f->light_source)
+			applyFacesShading(c2, v3f(dir.X, dir.Y, dir.Z));
 
-				u16 indices[] = {0,1,2,2,3,0};
-				// Add to mesh collector
-				collector.append(*current_tile, vertices, 4, indices, 6);
-			}
 
-			/*
-				Generate top side, if appropriate
-			*/
+		// The face at Z+
+		video::S3DVertex vertices[4] = {
+			video::S3DVertex(-BS/2,-BS/2,BS/2, dir.X,dir.Y,dir.Z, c2, 1,1),
+			video::S3DVertex(BS/2,-BS/2,BS/2, dir.X,dir.Y,dir.Z, c2, 0,1),
+			video::S3DVertex(BS/2,BS/2,BS/2, dir.X,dir.Y,dir.Z, c2, 0,0),
+			video::S3DVertex(-BS/2,BS/2,BS/2, dir.X,dir.Y,dir.Z, c2, 1,0),
+		};
 
-			if(top_is_same_liquid == false)
-			{
-				video::S3DVertex vertices[4] =
-				{
-					video::S3DVertex(-BS/2,0,BS/2, 0,0,0, c1, 0,1),
-					video::S3DVertex(BS/2,0,BS/2, 0,0,0, c1, 1,1),
-					video::S3DVertex(BS/2,0,-BS/2, 0,0,0, c1, 1,0),
-					video::S3DVertex(-BS/2,0,-BS/2, 0,0,0, c1, 0,0),
-				};
-
-				// To get backface culling right, the vertices need to go
-				// clockwise around the front of the face. And we happened to
-				// calculate corner levels in exact reverse order.
-				s32 corner_resolve[4] = {3,2,1,0};
-
-				for(s32 i=0; i<4; i++)
-				{
-					//vertices[i].Pos.Y += liquid_level;
-					//vertices[i].Pos.Y += neighbor_levels[v3s16(0,0,0)];
-					s32 j = corner_resolve[i];
-					vertices[i].Pos.Y += corner_levels[j];
-					if (data->m_smooth_lighting)
-						vertices[i].Color = blendLight(frame, vertices[i].Pos, tile_liquid.color);
-					vertices[i].Pos += intToFloat(p, BS);
-				}
+		// Rotations in the g_6dirs format
+		if(j == 0) // Z+
```

Comment:
ugh... how could github show such small diff part? (+2-2 lines)
@SmallJoker `NDT_GLASSLIKE` is on the queue.